cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -143,6 +156,7 @@ func (csr *ClusterStateRegistry) UpdateNodes(nodes []*apiv1.Node, currentTime ti
 	csr.nodes = nodes
 	csr.perNodeGroupReadiness, csr.totalReadiness = csr.calculateReadinessStats(currentTime)
 	csr.acceptableRanges = csr.calculateAcceptableRanges(targetSizes)
+	csr.incorrectNodeGroupSizes = csr.calculateIncorrectNodeGroupSizes(currentTime)
```

Comment:
The paradigm here is strange (lines 157-159): a.b = a.calculateB(). In my opinion a.updateB() would be much more clearer.